test/models/residence_test.rb

```diff
@@ -10,4 +10,26 @@ def test_valid
     assert residence.valid?, residence.errors.messages
   end
 
+  test 'should display homeless instead of residence if address is homeless or none' do 
+    addr = addresses :homeless
+    assert_equal addr.full, 'Homeless'
+
+    # tests 'currently homeless'
+    residence = Residence.new address_id: addr.id, current: true
+    assert_match /homeless/i, residence.to_s
```

Comment:
Actually, for these test cases, it makes sense to just test the whole string instead of doing regex comparisons.  That way, you verify that "current" is present in one case and absent in another.  Do assert_equal "Currently homeless", and so on for the other assertions.